pkg/segment/set.go

```diff
@@ -399,12 +429,20 @@ func (s *Set) Remove(seg Iterator) GapIterator {
 		// overlap.
 		seg.SetRangeUnchecked(victim.Range())
 		seg.SetValue(victim.Value())
+		// Need to update the sibling's maxGap in this case.
+		sibling := seg.NextSegment()
+		sibling.node.updateMaxGapLeaf()
 		return s.Remove(victim).NextGap()
 	}
 	copy(seg.node.keys[seg.index:], seg.node.keys[seg.index+1:seg.node.nrSegments])
 	copy(seg.node.values[seg.index:], seg.node.values[seg.index+1:seg.node.nrSegments])
 	Functions{}.ClearValue(&seg.node.values[seg.node.nrSegments-1])
 	seg.node.nrSegments--
+	// Update maxGap bottom up from the victim node.
```

Comment:
This actually can't be optimized because we should also consider the case where this function is called to remove the victim. In this case, seg.NextGap is ill defined because seg.NextSeg has the same range as seg.

Reply:
Yes I have tested it and it won't work properly with the optimization. 

I thought I have replied this thread but I couldn't find it :-(